### PR TITLE
added crude support for overriding the stream_ip

### PR DESCRIPTION
**Pull request by robbert229**
This PR is meant to solve a problem that I am encountering where the stream_ip that MA currently uses is the wrong ip address. Specifically I am running HA inside of kubernetes which creates its own internal ip addresses not accessable by the outside world.

I am not sure if this is the correct solution, but it worked for me. This PR is wip. Please let me know if there is some simple setting in HA that I should have tuned instead of doing this.

### CODE REVIEW

File: custom_components/mass/__init__.py [hunk 1]
```diff
@@ -95,7 +96,14 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
                 path=conf.get(CONF_FILE_DIRECTORY),
             )
         )
+
```
Comment (robbert229) [inline custom_components/mass/__init__.py:99]: This is where the heart of the changes are. We optionally override the stream_ip if a specific override has been given in the configuration of MA.

**Comment by marcelveldt** [comment]
This really should not be needed, as discussed in discord.
You will run into other trouble too if the ip couldn't be derived from the internal HA url.

This is just a patch at the wrong place and instead we should perhaps look why the 'hass.config.api.local_ip' value is wrong in your situation.

Normally if a user just fills in the correct internal and external URL's, this local_ip value is also correct.
Other stuff like TTS messages also rely on these URL's to be set correctly.


Can you add a small log statement to see what the value of hass.config.api.local_ip is in your situation ?
And can you also share the URL's you've filled in the HA config ?

If you're not comfortable in sharing the URL's here, we may also continue this conversation in a PM on discord.

**Comment by robbert229** [comment]
In my case the `hass.config.api.local_ip` resolves to 10.42.3.62.
My internal_url and external_url are both `https://homeassistant.infra.lab.johnrowley.co` both of these are A records that point to a loadbalancer which does tls termination. I used `get_url` to verify that both the internal and external urls are generated correctly. 

```
[31m2022-06-22 19:31:11 ERROR (Thread-16) [homeassistant.components.cast.media_player] Failed to cast media http://10.42.3.62:8095/89e478018eca4e5ebe22c174b93618eb.flac. Please make sure the URL is: Reachable from the cast device and either a publicly resolvable hostname or an IP address[0m
[31m2022-06-22 19:32:02 ERROR (Thread-16) [homeassistant.components.cast.media_player] Failed to cast media http://10.42.3.62:8095/cbaf31ea79734ca4af1ea848dcc3de95.flac. Please make sure the URL is: Reachable from the cast device and either a publicly resolvable hostname or an IP address[0m
```

When I stream local media using the built in media player it looks like it uses the internal/external urls correctly, hence my original assumption that the issue lies with music assistant.

**Comment by marcelveldt** [comment]
Wow, that is a pretty complex setup for hosting HA in your home :-)
I'd like to use an approach as universal as possible. It's a component that is an extension from HA so HA should be the one to determinate the IP/dns names to use. Based on your comment in discord I now learned that the 'hass.config.api.local_ip' is not actually resolving the 'internal address' into an IP but it's using logic to grab the IP itself from the interfaces.

We could create a small helper function in the MA integration code to resolve the 'internal url' into an IP ?

**Comment by marcelveldt** [comment]
So 'hass.config.api.local_ip' should be replaced with 'get_local_ip_from_internal_url'

**Comment by robbert229** [comment]
`get_local_ip_from_internal_url` sounds like it would solve my exact issue. Thanks for figuring this out :)

**Pull request closed without merging**